api/layer/container.go

```diff
@@ -207,6 +207,43 @@ func (n *layer) waitEACLPresence(ctx context.Context, cid *cid.ID, table *eacl.T
 		return fmt.Errorf("couldn't marshal eacl: %w", err)
 	}
 
+	return waitFor(ctx, params, func(ctx context.Context) bool {
+		eaclTable, err := n.pool.GetEACL(ctx, cid)
+		if err == nil {
+			got, err := eaclTable.Marshal()
+			if err == nil && bytes.Equal(exp, got) {
+				return true
+			}
+		}
+		return false
+	})
+}
+
+func (n *layer) deleteContainer(ctx context.Context, cid *cid.ID) error {
+	var sessionToken *session.Token
+	boxData, err := GetBoxData(ctx)
+	if err == nil {
+		sessionToken = boxData.Gate.SessionTokenForDelete()
+	}
+
+	if err = n.pool.DeleteContainer(ctx, cid, pool.WithSession(sessionToken)); err != nil {
+		return err
+	}
+
+	return n.waitForContainerRemoved(ctx, cid, defaultWaitParams())
+}
+
+func (n *layer) waitForContainerRemoved(ctx context.Context, cid *cid.ID, params *waitParams) error {
+	return waitFor(ctx, params, func(ctx context.Context) bool {
+		_, err := n.pool.GetContainer(ctx, cid)
+		if err != nil && strings.Contains(err.Error(), "container not found") {
```

Comment:
Lets add `TODO: (neofs-s3-gw#367) handle NeoFS API status error`?

Also we can just `return` this statement w/o `if` block.

Reply:
Currently, it doesn't work. Do we need use neofs-node version other than v0.27.5?

Reply:
I know, meant just to add the TODO comment.